tools/VscPreviewExtension/src/Common/tempPreviewFileProcessor.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { ExtensionContext, window, workspace } from "vscode";
+import * as path from "path";
+import * as fs from "fs";
+
+import * as ConstVariables from "../ConstVariables/commonVariables";
+import { TempPreviewFileInformation } from "./tempPreviewFileInformation";
+
+export class TempPreviewFileProcessor {
+    // TODO: Write\Delete temp preview file at client side instead of server side.
+    public static initializeTempFileInformation(context: ExtensionContext, navigationPort: string, config) {
+        let workspacePath = workspace.rootPath;
+        let editor = window.activeTextEditor;
+        let doc = editor.document;
+        let fileName = doc.fileName;
+        let rootPathLength = workspacePath.length;
+        let relativePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
```

Comment:
>fileName [](start = 27, length = 8)

several places use extract file name or file name without ext, consider put it in utility